src/main/java/org/folio/inventory/resources/AbstractInstances.java

```diff
@@ -109,61 +109,161 @@ protected void updateInstanceRelationships(Instance instance,
+    List<CompletableFuture<Response>> allFutures,
+    Map<String, T> existingObjects, Map<String, T> updatingObjects) {
+
+    createOrEdit(resourceClient, allFutures, existingObjects, updatingObjects);
+    delete(resourceClient, allFutures, existingObjects, updatingObjects);
+  }
+
+  private <T> void delete(CollectionResourceClient resourceClient,
+   List<CompletableFuture<Response>> allFutures, Map<String, T> existingObjects,
+   Map<String, T> updatingObjects) {
+
+    existingObjects.keySet().stream()
+      .filter(key -> !updatingObjects.containsKey(key)).forEach(existingKey -> {
+      CompletableFuture<Response> newFuture = new CompletableFuture<>();
+      allFutures.add(newFuture);
+      resourceClient.delete(existingKey, newFuture::complete);
+    });
+  }
+
+  private <T> void createOrEdit(CollectionResourceClient resourceClient,
+                                List<CompletableFuture<Response>> allFutures, Map<String, T> existingObjects,
+                                Map<String, T> updatingObjects) {
+
+    updatingObjects.keySet().forEach(updatingKey -> {
+      T object = updatingObjects.get(updatingKey);
+      if (existingObjects.containsKey(updatingKey)) {
+        if (!updatingObjects.get(updatingKey).equals(existingObjects.get(updatingKey))) {
+          CompletableFuture<Response> newFuture = new CompletableFuture<>();
+          allFutures.add(newFuture);
+          resourceClient.put(updatingKey, object, newFuture::complete);
+        }
+      } else {
+        CompletableFuture<Response> newFuture = new CompletableFuture<>();
+        allFutures.add(newFuture);
+        resourceClient.post(object, newFuture::complete);
+      }
+    });
+  }
+
+  private Map<String, PrecedingSucceedingTitle> getUpdatingPrecedingSucceedingTitles(Instance instance) {
+    Map<String, PrecedingSucceedingTitle> updatingPrecedingSucceedingTitles = new HashMap();
+    if (instance.getPrecedingTitles() != null) {
+      instance.getPrecedingTitles().forEach(parent -> {
+        String id = (parent.id == null ? UUID.randomUUID().toString() : parent.id);
+        PrecedingSucceedingTitle precedingSucceedingTitle = new PrecedingSucceedingTitle(
+          id,
+          parent.precedingInstanceId,
+          instance.getId(),
+          parent.title,
+          parent.hrid,
+          parent.identifiers);
+
+        updatingPrecedingSucceedingTitles.put(id, precedingSucceedingTitle);
+      });
+    }
+    if (instance.getSucceedingTitles() != null) {
+      instance.getSucceedingTitles().forEach(child -> {
+        String id = (child.id == null ? UUID.randomUUID().toString() : child.id);
+        updatingPrecedingSucceedingTitles.put(id,
+          new PrecedingSucceedingTitle(
+            id,
+            instance.getId(),
+            child.succeedingInstanceId,
+            child.title,
+            child.hrid,
+            child.identifiers));
+      });
+    }
+    return updatingPrecedingSucceedingTitles;
+  }
```

Comment:
Personally, I might include some whitespace between some of these statements to separate the various blocks of logic to aid comprehension.

Reply:
Divided into two methods